### PR TITLE
Hide reusable blocks from story editor global inserter.

### LINKED ISSUE #3105: Reusable blocks shouldn't show up in the Global Inserter
Reusable blocks might contain blocks that are not allowed within AMP Story Pages and should not be allowed.

We could filter the post query so that on AMP Story post type it would only include reusable blocks which have `TEMPLATES_TERM` and `TEMPLATES_TAXONOMY` assigned (the opposite of what we're doing right now for non-Story post types).

### PR DESCRIPTION
**Pull request by miina**
Fixes #3105.

~Would be good to get #3166 merged first.~ DONE

### CODE REVIEW

**Comment by swissspidy** [comment]
Looks like merging #3166 caused a conflict

**Comment by miina** [comment]
Yep, updating in a sec, that was expected.

**Comment by swissspidy** [comment]
Thanks. Changes look good at first glance.

**Comment by miina** [comment]
Eh, updated the comment for clarity as well, now should ba actually ready for review.

File: tests/e2e/specs/stories-editor/story-templates.js [hunk 1]
```diff
@@ -15,9 +15,59 @@ import {
...
+ */
+async function removeAllReusableBlocks() {
+	await visitAdminPage( 'edit.php', 'post_type=wp_block' );
+
+	// Delete all reusable blocks to restore clean state.
+	const selector = '#cb-select-all-1';
+	const actionsSelector = '#bulk-action-selector-top';
+
+	await page.click( selector );
+	await page.select( actionsSelector, 'trash' );
```
Comment (swissspidy) [inline tests/e2e/specs/stories-editor/story-templates.js]: The `should hide story templates from the reusable blocks management screen` test is currently failing at this point:

`Error: failed to find element matching selector "#bulk-action-selector-top"`

Comment (swissspidy) [reply tests/e2e/specs/stories-editor/story-templates.js]: https://travis-ci.org/ampproject/amp-wp/jobs/580162773#L923-L933

**Pull request merged**